packages/cli/src/utils/build.js

```diff
@@ -71,171 +70,396 @@ const requiredFiles = async ({ cwd, entryPoints }) => {
+  const relPaths = Array.from(
+    new Set([
+      // Files found by esbuild and their package.json files
+      ...expandRequiredFiles(
+        workingDir,
+        await findRequiredFiles(workingDir, entryPoints),
+      ),
+      // Files matching includeInBuild and other preset patterns
+      ...itermap(
+        (entry) =>
+          path.relative(workingDir, path.join(entry.parentPath, entry.name)),
+        walkDirWithPatterns(workingDir, appConfig?.includeInBuild),
+      ),
+    ]),
+  ).sort();
+
+  const workspaceRoot = (await findWorkspaceRoot(workingDir)) || workingDir;
+
+  if (workspaceRoot !== workingDir) {
+    const appDirRelPath = path.relative(workspaceRoot, workingDir);
+    const linkNames = ['zapierwrapper.js', 'index.js'];
+    for (const name of linkNames) {
+      zip.symlink(name, path.join(appDirRelPath, name), 0o644);
+    }
 
-  const [dumbPaths, smartPaths, appConfig] = await Promise.all([
-    listFiles(dir),
-    requiredFiles({ cwd: dir, entryPoints }),
-    getLinkedAppConfig(dir).catch(() => ({})),
-  ]);
+    const filenames = ['package.json', 'definition.json'];
+    for (const name of filenames) {
+      const absPath = path.resolve(workingDir, name);
+      zip.file(absPath, { name, mode: 0o644 });
+    }
+  }
+
+  // Write required files to the zip
+  for (const relPath of relPaths) {
+    const absPath = path.resolve(workingDir, relPath);
+    const nameInZip = path.relative(workspaceRoot, absPath);
+    if (nameInZip === 'package.json' && workspaceRoot !== workingDir) {
+      // Ignore workspace root's package.json
+      continue;
+    }
+    zip.file(absPath, { name: nameInZip, mode: 0o644 });
+  }
+
+  // Next, find all symlinks that are either: (1) immediate children of any
+  // node_modules directory, or (2) located one directory level below a
+  // node_modules directory. (1) is for the case of node_modules/package_name.
+  // (2) is for the case of node_modules/@scope/package_name.
+  const nodeModulesDirs = new Set();
+  for (const relPath of relPaths) {
+    const nmDir = getNearestNodeModulesDir(workingDir, relPath);
+    if (nmDir) {
+      nodeModulesDirs.add(nmDir);
+    }
+  }
+
+  for (const relNmDir of nodeModulesDirs) {
+    const absNmDir = path.resolve(workingDir, relNmDir);
+    const symlinks = iterfilter(
+      (entry) => {
+        // Only include symlinks that are not in node_modules/.bin directories
+        return (
+          entry.isSymbolicLink() &&
+          !entry.parentPath.endsWith(`${path.sep}node_modules${path.sep}.bin`)
+        );
+      },
+      walkDirLimitedLevels(absNmDir, 2),
+    );
+    for (const symlink of symlinks) {
+      const absPath = path.resolve(
+        workingDir,
+        symlink.parentPath,
+        symlink.name,
+      );
+      const nameInZip = path.relative(workspaceRoot, absPath);
+      const targetInZip = path.relative(
+        symlink.parentPath,
+        fs.realpathSync(absPath),
```

Comment:
```suggestion
        await fs.realpath(absPath),
```
For non-blocking IO performance